api/envoy/config/core/v3/protocol.proto

```diff
@@ -651,6 +651,13 @@ message Http3ProtocolOptions {
 message SchemeHeaderTransformation {
   oneof transformation {
     // Overwrite any Scheme header with the contents of this string.
+    // If set, takes precedence over match_upstream.
     string scheme_to_overwrite = 1 [(validate.rules).string = {in: "http" in: "https"}];
   }
+
+  // Set the Scheme header to match the upstream transport protocol. For example, should a
+  // request be sent to the upstream over TLS, the scheme header will be set to "https". Should the
+  // request be sent over plaintext, the scheme header will be set to "http".
+  // If scheme_to_overwrite is set, this field is not used.
+  bool match_upstream = 2;
```

Comment:
Not too familiar with how validation of configs is done but my understanding is it uses `protoc-gen-validate` and I'm not sure if that schema supports cross-field rules?

Reply:
Should we just use `oneof` then since this new field doesn't take precedence so if a user wants to use this field they have to not set the other field?

Reply:
> Edit: Actually even then it probably isn't a good idea as there may be a new field which takes precedence in the future.

Yeah, that's the challenge with allowing oneof in this case.
I think that explicitly stating the priorities, and (optionally) emitting a warning when the two fields are set together is a decent compromise.

Reply:
I think in this case I should keep the PR as is w.r.t. the API and the emitting the warning. Think I can get a sign-off to pass the CI? Or is it blocked on a review meeting?